torch/_higher_order_ops/auto_functionalize.py

```diff
@@ -149,16 +209,22 @@ def read_single_view(prefix):
         elif f"{prefix}_alias" in kwargs:
             get_arg(f"{prefix}_alias")
             return ViewInfo(base_index, all_bases[base_index], is_alias=True)
-        elif f"{prefix}_size" not in kwargs:
-            assert f"{prefix}_stride" not in kwargs
-            assert f"{prefix}_storage_offset" not in kwargs
-            # This means that the argument is the base tensor
-            return ViewInfo(base_index, all_bases[base_index], is_view=False)
-        else:
+        elif f"{prefix}_storage_offset" in kwargs:
+            # The view is regenerated using as_strided.
             size = get_arg(f"{prefix}_size")
             stride = get_arg(f"{prefix}_stride")
             storage_offset = get_arg(f"{prefix}_storage_offset")
-            return ViewInfo(base_index, size, stride, storage_offset)
+            return ViewInfo(
+                base_index, as_strided_info=(size, stride, storage_offset), is_view=True
+            )
+        elif f"{prefix}_slice_dim" in kwargs:
+            dim = get_arg(f"{prefix}_slice_dim")
```

Comment:
(no action required) I also don't like how the serialization scheme is going. Maybe leave a comment that "if you want to support auto_functionalized_v2 for export -- please just implement auto_functionalized_v3"